src/bls12_381/ec.rs

```diff
@@ -1653,6 +1733,17 @@ pub mod g2 {
         ::tests::curve::curve_tests::<G2>();
     }
 
+    #[test]
+    fn g2_test_hash() {
+        let mut rng = XorShiftRng::from_seed([0x5dbe6259, 0x8d313d76, 0x3237db17, 0xe5bc0654]);
```

Comment:
good catch, there's a problem in here already.

Definition 2 states extends the encoding from Fq* to Fq with f(0).y = sqrt(1 + b). However, in our case 1+b is not a quadratic residue. Bummer.